### PR TITLE
Feature FSI MPI MVQN recursive

### PR DESCRIPTION
**Pull request by rubenzorrilla**
This PR adds the MPI version of the recursive MVQN convergence accelerator.

### CODE REVIEW

File: applications/FSIapplication/custom_utilities/mvqn_recursive_convergence_accelerator.hpp [hunk 1]
```diff
@@ -14,17 +14,14 @@
...
-//~ #include <numeric>
 
 /* Project includes */
 #include "includes/define.h"
 #include "includes/variables.h"
 #include "includes/ublas_interface.h"
 #include "convergence_accelerator.hpp"
 
-#include "custom_utilities/qr_utility.h"            //QR decomposition utility used in matrix inversion.
+#include "../FSIapplication/custom_utilities/qr_utility.h"            //QR decomposition utility used in matrix inversion.
```
Comment (RiccardoRossi) [inline applications/FSIapplication/custom_utilities/mvqn_recursive_convergence_accelerator.hpp]: the ../FSIapplication should not be needed.

also i think that the qr_utility shall be moved to the core

File: applications/FSIapplication/custom_utilities/mvqn_recursive_convergence_accelerator.hpp [hunk 2]
```diff
@@ -175,20 +173,30 @@ class JacobianEmulator
     {
         KRATOS_TRY;
 
+        // std::cout << std::endl;
```
Comment (RiccardoRossi) [inline applications/FSIapplication/custom_utilities/mvqn_recursive_convergence_accelerator.hpp]: i just suggest cleaning up the comments

File: applications/FSIapplication/custom_utilities/mvqn_recursive_convergence_accelerator.hpp [hunk 3]
```diff
@@ -434,14 +443,15 @@ class MVQNRecursiveJacobianConvergenceAccelerator: public ConvergenceAccelerator
...
-            "solver_type" : "MVQN",
+            "solver_type" : "MVQN_recursive",
             "w_0"         : 0.825,
             "buffer_size" : 10
         }
         )");
 
         rConvAcceleratorParameters.ValidateAndAssignDefaults(mvqn_recursive_default_parameters);
 
+        mColumnCutOffRelTol = 1e-8;
```
Comment (RiccardoRossi) [inline applications/FSIapplication/custom_utilities/mvqn_recursive_convergence_accelerator.hpp]: shouldn't this be in the parameters with a default value?

File: applications/FSIapplication/tests/convergence_accelerator_spring_MPI_test.py [hunk 4]
```diff
@@ -172,11 +166,11 @@ def test_aitken_accelerator(self,force1,force2,solution):
         residual = self.ComputeResidual(top_part,x_guess,force1,force2)
         res_norm = self.ComputeResidualNorm(residual)
 
-        while (nl_it <= self.aitken_iterations):
+        while (nl_it <= self.accelerator_iterations):
 
             print(mpi.rank,": Iteration: ", nl_it," residual norm: ", res_norm, file=sys.stderr)
 
-            if res_norm > self.aitken_tolelance:
+            if res_norm > self.accelerator_tolelance:
```
Comment (RiccardoRossi) [inline applications/FSIapplication/tests/convergence_accelerator_spring_MPI_test.py]: bad spelling

File: applications/FSIapplication/tests/convergence_accelerator_spring_MPI_test.py [hunk 5]
```diff
@@ -264,8 +269,8 @@ def test_aitken_accelerator_ghost_nodes(self):
         self.print_gid_output = False
 
         # relax tolerance requirements to force differences between processors
-        self.aitken_tolelance = 1e-2
-        self.aitken_iterations = 10
+        self.accelerator_tolelance = 1e-2
```
Comment (RiccardoRossi) [inline applications/FSIapplication/tests/convergence_accelerator_spring_MPI_test.py]: bad spelling

File: applications/trilinos_application/custom_python/add_custom_utilities_to_python.cpp [hunk 6]
```diff
@@ -163,48 +165,56 @@ void  AddCustomUtilitiesToPython()
...
+        .def("ComputeFluidInterfaceMeshVelocityResidualNorm",&TrilinosPartitionedFSIUtilities3DType::ComputeFluidInterfaceMeshVelocityResidualNorm)
+        .def("ComputeAndPrintFluidInterfaceNorms", &TrilinosPartitionedFSIUtilities3DType::ComputeAndPrintFluidInterfaceNorms)
+        .def("ComputeAndPrintStructureInterfaceNorms", &TrilinosPartitionedFSIUtilities3DType::ComputeAndPrintStructureInterfaceNorms)
+        .def("CheckCurrentCoordinatesFluid", &TrilinosPartitionedFSIUtilities3DType::CheckCurrentCoordinatesFluid)
+        .def("CheckCurrentCoordinatesStructure", &TrilinosPartitionedFSIUtilities3DType::CheckCurrentCoordinatesStructure)
+        ;
 
 
     // Convergence accelerators (from FSIApplication)
     typedef ConvergenceAccelerator<TrilinosSparseSpaceType> TrilinosConvergenceAccelerator;
```
Comment (RiccardoRossi) [inline applications/trilinos_application/custom_python/add_custom_utilities_to_python.cpp:195]: this means that the FSI app now depends on mpi? :-o

File: applications/trilinos_application/custom_utilities/trilinos_mvqn_recursive_convergence_accelerator.hpp [hunk 7]
```diff
@@ -0,0 +1,769 @@
+//    |  /           |
```
Comment (RiccardoRossi) [inline applications/trilinos_application/custom_utilities/trilinos_mvqn_recursive_convergence_accelerator.hpp:1]: ...yes it does :-(. We will need to cleanup this, although at a later stage

**Review by RiccardoRossi — COMMENTED** [review]
is trilinos optional at least?

**Review by RiccardoRossi — APPROVED** [review]

File: applications/FSIapplication/custom_utilities/mvqn_recursive_convergence_accelerator.hpp (hunk 1 above)
Comment (rubenzorrilla) [reply applications/FSIapplication/custom_utilities/mvqn_recursive_convergence_accelerator.hpp]: Done. The QR utility has been moved to the core.

File: applications/FSIapplication/custom_utilities/mvqn_recursive_convergence_accelerator.hpp (hunk 2 above)
Comment (rubenzorrilla) [reply applications/FSIapplication/custom_utilities/mvqn_recursive_convergence_accelerator.hpp]: Done. I took the chance to formatting the file to the new standards.

File: applications/FSIapplication/custom_utilities/mvqn_recursive_convergence_accelerator.hpp (hunk 3 above)
Comment (rubenzorrilla) [reply applications/FSIapplication/custom_utilities/mvqn_recursive_convergence_accelerator.hpp]: Should be. Added to both constructors.

File: applications/FSIapplication/tests/convergence_accelerator_spring_MPI_test.py (hunk 4 above)
Comment (rubenzorrilla) [reply applications/FSIapplication/tests/convergence_accelerator_spring_MPI_test.py]: Corrected.

File: applications/FSIapplication/tests/convergence_accelerator_spring_MPI_test.py (hunk 5 above)
Comment (rubenzorrilla) [reply applications/FSIapplication/tests/convergence_accelerator_spring_MPI_test.py]: Corrected.

File: applications/trilinos_application/custom_utilities/trilinos_mvqn_recursive_convergence_accelerator.hpp (hunk 7 above)
Comment (rubenzorrilla) [reply applications/trilinos_application/custom_utilities/trilinos_mvqn_recursive_convergence_accelerator.hpp:1]: Yes we do...

**Pull request merged**